### DISCUSSION TITLE
Why Laravel Herd doesn't include gettext extension out of the box?§

### DISCUSSION THREAD

**Discussion by toghrul-nasirli** [beyondcode/herd-community#433] [Q&A]
I had to uninstall `Herd` because I couldn't find a way to install the `gettext` extension. Herd's official documentation says install extensions via `pecl`, but the problem is that `pecl` does not have the `gettext` extension in its library. So I had to install it manually and eventually gave up. But seriously, `gettext` is a widely needed extension and I don't know why `Herd` doesn't include it out of the box.

And how do we install other extensions if they are not available in `pecl`?

**Comment by tandasima** [comment]
Hi @toghrul-nasirli to install the gettext extension in Laravel Herd you'll need to do the following.
1. Install php via home-brew: `brew install php`
2. Download the `gettext` image from `https://packages.macports.org/php83-gettext/`. Ofcourse, match the same version like one you have installed via Laravel Herd. The link I provided in for `php 8.3.x`. Also, if you are using Mac M1/2... chip you should download the latest `arm64` one. Eg. [php83-gettext-8.3.8_0.darwin_21.arm64.tbz2](https://packages.macports.org/php83-gettext/php83-gettext-8.3.8_0.darwin_21.arm64.tbz2) as of this writing.
3. Put the download anywhere it makes sense, eg. in `~/Herd/`. Also, if you are using a Mac you will want to open the `gettext` at least once so you can allow it to be opened by the system, unless herd will fail to load it.
4. Following instructions from Herd, [Adding additional extensions to php.ini](https://herd.laravel.com/docs/1/advanced-usage/additional-extensions), add the gettext.so path to the relevant file.
5. Quit herd and run `herd start`. Herd should start successfully. Sometimes it'd complain `libintl.8.dylib` file was not found. If this happens, you can create a symbolic link from homebrew's to the location where it is trying to find it.
   - Find it by running `find /opt/homebrew -name "libintl.8.dylib"`.
   - If it is expecting it to be at `/usr/local/lib/` and the above command finds the home-brew one at `/opt/homebrew/opt/gettext/lib/libintl.8.dylib` you can create the symbolic link by running `sudo ln -s /opt/homebrew/opt/gettext/lib/libintl.8.dylib /usr/local/lib/libintl.8.dylib`

I hope this helps.
